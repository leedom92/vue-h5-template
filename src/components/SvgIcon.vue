<template>
  <svg aria-hidden="true">
    <use :href="symbolId" :fill="color" />
  </svg>
</template>

<script setup>
const props = defineProps({
  name: {
    type: String,
    required: true,
  },
  prefix: {
    type: String,
    default: 'icon',
  },
  color: {
    type: String,
    default: '#000',
  },
})

const symbolId = computed(() => `#${props.prefix}-${props.name}`)
</script>
