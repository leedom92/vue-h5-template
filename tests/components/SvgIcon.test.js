import SvgIcon from '@/components/SvgIcon.vue'

describe('SvgIcon.vue', () => {
  it('name', () => {
    const wrapper = mount(SvgIcon, {
      props: {
        name: 'leedom',
      },
    })
    expect(wrapper.find('use').attributes().href).toBe('#icon-leedom').toMatchSnapshot()
  })

  it('color', () => {
    const wrapper = mount(SvgIcon, {
      props: {
        name: 'leedom',
        color: '#666',
      },
    })
    expect(wrapper.find('use').attributes().fill).toBe('#666').toMatchSnapshot()
  })
})
