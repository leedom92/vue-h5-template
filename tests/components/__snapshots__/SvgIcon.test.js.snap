// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SvgIcon.vue > color 1`] = `"#666"`;

exports[`SvgIcon.vue > name 1`] = `"#icon-leedom"`;
